src/nomad_measurements/xrd/nx.py

```diff
@@ -0,0 +1,222 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import TYPE_CHECKING
+from pynxtools.nomad.dataconverter import populate_nexus_subsection
+from pynxtools import dataconverter
+
+
+if TYPE_CHECKING:
+    from nomad.datamodel.datamodel import EntryArchive
+
+
+# source_peak_wavelength
+def connect_concepts(template, archive: 'EntryArchive', scan_type: str):
+    """Connect the concepts between ELNXrayDiffraction and NXxrd_pan schema.
+    
+    Args:
+        template (Template): The pynxtools template, a inherited class from python dict.
+        archive (EntryArchive): Nomad archive contains secttions, subsections and quantities.
+        scan_type (str): Name of the scan type such as line and RSM.
+    """
+
+    # Genneral concepts
+    template['/ENTRY[entry]/definition'] = 'NXxrd_pan'
+
+    try:
+        template['/ENTRY[entry]/method'] = archive.data.method
+    except AttributeError:
+        pass
+
+    try:
+        template['/ENTRY[entry]/measurement_type'] = (
+            archive.data.diffraction_method_name
+        )
+    except AttributeError:
+        pass
+
+    try:
+        template['/ENTRY[entry]/experiment_result/intensity'] = archive.data.results[0].intensity.magnitude
+    except AttributeError:
+        pass
+
+    try:
+        template['/ENTRY[entry]/experiment_result/two_theta'] = archive.data.results[0].two_theta.magnitude
+        template['/ENTRY[entry]/experiment_result/two_theta/@units'] = archive.data.results[0].two_theta.units.__str__()
```

Comment:
I just ran the linter and it suggests using `str()` method instead of the dunder call `.__str__()`

here's some docu for this https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/unnecessary-dunder-call.html

```suggestion
        template['/ENTRY[entry]/experiment_result/two_theta/@units'] = str(archive.data.results[0].two_theta.units)
```

Reply:
It would be nice to stick with the lining suggestions. Please make similar changes in other places too.